### PR TITLE
[luci-interpreter] Use optimized_ops Gather

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit uses optimized_ops Gather instead of reference_ops.

Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Related comment: https://github.com/Samsung/ONE/pull/8255#discussion_r779323704

### CODE REVIEW

**Comment by jyoungyun** [comment]
Test passed
```
$ ./build/compiler/luci-interpreter/src/kernels/luci_interpreter_kernels_test --gtest_filter=Gather*
Running main() from gtest_main.cc
Note: Google Test filter = Gather*
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from GatherTest
[ RUN      ] GatherTest.Simple
[       OK ] GatherTest.Simple (1 ms)
[ RUN      ] GatherTest.Simple_NEG
[       OK ] GatherTest.Simple_NEG (0 ms)
[----------] 2 tests from GatherTest (1 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (1 ms total)
[  PASSED  ] 2 tests.
```

File: compiler/luci-interpreter/pal/linux/PALGather.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+#ifndef LUCI_INTERPRETER_PAL_GATHER_H
+#define LUCI_INTERPRETER_PAL_GATHER_H
+
+#include <tensorflow/lite/kernels/internal/optimized/optimized_ops.h>
+
+namespace luci_interpreter_pal
+{
+template <typename T, typename CoordsT = int32>
```
Comment (jinevening) [inline compiler/luci-interpreter/pal/linux/PALGather.h:24]: We may not use the default value for arguments (this is for removing implicit behavior).

```suggestion
template <typename T, typename CoordsT>
```

**Review by jinevening — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
I'd like to land this optmized kernel after my full draft has landed. Is that OK?

**Comment by seanshpark** [comment]
@jyoungyun please rebase to master

**Comment by jyoungyun** [comment]
I updated it.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**